### PR TITLE
optimize layers in dockerfile

### PR DESCRIPTION
**Pull request by mashehu**
This should allow faster opening times in codespaces and on gitpod 🤞🏻

### CODE REVIEW

File: nf_core/gitpod/gitpod.Dockerfile [hunk 1]
```diff
@@ -8,7 +8,7 @@ USER root
 # Install util tools.
 # software-properties-common is needed to add ppa support for Apptainer installation
 RUN apt-get update --quiet && \
-    apt-get install --quiet --yes \
+    apt-get install --quiet --yes --no-install-recommends \
```
Comment (mirpedrol) [inline nf_core/gitpod/gitpod.Dockerfile:11]: sounds scary 😮 are we sure we don't want to install any recommended package?

File: nf_core/gitpod/gitpod.Dockerfile [hunk 2]
```diff
@@ -47,23 +46,20 @@ RUN conda config --add channels defaults && \
     conda config --add channels conda-forge && \
     conda config --set channel_priority strict && \
     conda install --quiet --yes --name base \
-    mamba \
```
Comment (mirpedrol) [inline nf_core/gitpod/gitpod.Dockerfile:50]: I wonder if someone uses mamba to run tests?

**Review by mirpedrol — APPROVED** [review]
LGTM

File: nf_core/gitpod/gitpod.Dockerfile (hunk 2 above)
Comment (mashehu) [reply nf_core/gitpod/gitpod.Dockerfile:50]: we will find out 🙂

**Pull request merged**